src/routes/transactions/mappers/multisig-transaction.mapper.ts

```diff
@@ -0,0 +1,599 @@
+  NativeCoinTransferInfo,
+  TransferDirection,
+  TransferTransaction,
+} from '../entities/transfer-transaction.entity';
+
+@Injectable()
+export class MultisigTransactionMapper {
+  private readonly NULL_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  private static readonly SET_FALLBACK_HANDLER = 'setFallbackHandler';
+  private static readonly ADD_OWNER_WITH_THRESHOLD = 'addOwnerWithThreshold';
+  private static readonly REMOVE_OWNER = 'removeOwner';
+  private static readonly SWAP_OWNER = 'swapOwner';
+  private static readonly CHANGE_THRESHOLD = 'changeThreshold';
+  private static readonly CHANGE_MASTER_COPY = 'changeMasterCopy';
+  private static readonly ENABLE_MODULE = 'enableModule';
+  private static readonly DISABLE_MODULE = 'disableModule';
+  private static readonly SET_GUARD = 'setGuard';
+
+  private readonly SETTINGS_CHANGE_METHODS = [
+    MultisigTransactionMapper.SET_FALLBACK_HANDLER,
+    MultisigTransactionMapper.ADD_OWNER_WITH_THRESHOLD,
+    MultisigTransactionMapper.REMOVE_OWNER,
+    MultisigTransactionMapper.SWAP_OWNER,
+    MultisigTransactionMapper.CHANGE_THRESHOLD,
+    MultisigTransactionMapper.CHANGE_MASTER_COPY,
+    MultisigTransactionMapper.ENABLE_MODULE,
+    MultisigTransactionMapper.DISABLE_MODULE,
+    MultisigTransactionMapper.SET_GUARD,
+  ];
+
+  private readonly TRANSFER_METHOD = 'transfer';
+  private readonly TRANSFER_FROM_METHOD = 'transferFrom';
+  private readonly SAFE_TRANSFER_FROM_METHOD = 'safeTransferFrom';
+  private readonly ERC20_TRANSFER_METHODS = [
+    this.TRANSFER_METHOD,
+    this.TRANSFER_FROM_METHOD,
+  ];
+  private readonly ERC721_TRANSFER_METHODS = [
+    this.TRANSFER_METHOD,
+    this.TRANSFER_FROM_METHOD,
+    this.SAFE_TRANSFER_FROM_METHOD,
+  ];
+
+  constructor(
+    private readonly addressInfoHelper: AddressInfoHelper,
+    @Inject(ITokenRepository) private readonly tokenRepository: TokenRepository,
+  ) {}
+
+  async mapToTransactionSummary(
+    chainId: string,
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): Promise<TransactionSummary> {
+    const txStatus = this.mapTransactionStatus(transaction, safe);
+    const txInfo = await this.mapTransactionInfo(chainId, transaction, safe);
+
+    return {
+      id: `multisig_${transaction.safe}_${transaction.safeTxHash}`,
+      timestamp:
+        transaction?.executionDate?.getTime() ??
+        transaction?.submissionDate?.getTime(),
+      txStatus,
+      txInfo,
+      executionInfo: this.mapExecutionInfo(transaction, safe, txStatus),
+      // TODO: include safeAppInfo retrieval logic where needed
+    };
+  }
+
+  private async mapSettingsInfo(
+    chainId: string,
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): Promise<SettingsInfo | undefined> {
+    const { dataDecoded } = transaction;
+    switch (dataDecoded.method) {
+      case MultisigTransactionMapper.SET_FALLBACK_HANDLER:
+        return <SetFallbackHandler>{
+          type: 'SET_FALLBACK_HANDLER',
```

Comment:
I think we should set the `type` as a property in the class itself. Allowing `SetFallbackHandler` + separately setting the `label` would involve more work.

Reply:
Changed in [c464a50](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/c464a504861327618faa18ddc5c0662aace9d5a0)

Reply:
Explicit constructors were introduced in [c464a50](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/c464a504861327618faa18ddc5c0662aace9d5a0)